src/qibocal/protocols/drag.py

```diff
@@ -79,70 +80,87 @@ def _acquisition(
 
     data = DragTuningData(
         anharmonicity={
-            qubit: platform.qubits[qubit].anharmonicity * HZ_TO_GHZ for qubit in targets
+            qubit: platform.calibration.single_qubits[qubit].qubit.anharmonicity
+            * HZ_TO_GHZ
+            for qubit in targets
         }
     )
-    # define the parameter to sweep and its range:
-    # qubit drive DRAG pulse beta parameter
     beta_param_range = np.arange(params.beta_start, params.beta_end, params.beta_step)
 
     sequences, all_ro_pulses = [], []
     for beta_param in beta_param_range:
         sequence = PulseSequence()
         ro_pulses = {}
-        for qubit in targets:
-            RX_drag_pulse = platform.create_RX_drag_pulse(
-                qubit, start=0, beta=beta_param / data.anharmonicity[qubit]
-            )
-            RX_drag_pulse_minus = platform.create_RX_drag_pulse(
-                qubit,
-                start=RX_drag_pulse.finish,
-                beta=beta_param / data.anharmonicity[qubit],
-                relative_phase=np.pi,
-            )
-            ro_pulses[qubit] = platform.create_qubit_readout_pulse(
-                qubit, start=RX_drag_pulse_minus.finish
+        for q in targets:
+            natives = platform.natives.single_qubit[q]
+            qd_channel, qd_pulse = natives.RX()[0]
+            ro_channel, ro_pulse = natives.MZ()[0]
+
+            drag = replace(
+                qd_pulse,
+                envelope=Drag(
+                    rel_sigma=qd_pulse.envelope.rel_sigma,
+                    beta=beta_param / data.anharmonicity[q],
+                ),
             )
+            drag_negative = replace(drag, relative_phase=np.pi)
+
+            # TODO: here we can add pairs of this in a for loop
+            sequence.append((qd_channel, drag))
+            sequence.append((qd_channel, drag_negative))
+
```

Comment:
I think here we are missing a delay
```suggestion
sequence.append((ro_channel, Delay(duration=drag.duration + drag_negative.duration)))
```
so that the readout starts after the gates are applied.